### PR TITLE
fix: resource deletion and adoption for 3 controllers

### PR DESCRIPTION
**Pull request by ksdpmx**
### What this PR does / why we need it:

~resources like sa, svc and sts haven't been generated before calling `Delete()` which leads to the remaining finalizer for collector.~
resource deletion and adoption cleanup and fix for 3 controllers

### Which issue(s) this PR fixes:

Fixes #

### Does this PR introduced a user-facing change?

```release-note
None
```

### Additional documentation, usage docs, etc.:

```docs

```

### CODE REVIEW

**Comment by ksdpmx** [comment]
> "2023-06-06T03:10:30Z ERROR Reconciler error {"controller": "collector", "controllerGroup": "fluentbit.fluent.io", "controllerKind": "Collector", "Collector": {"name":"fluent-bit-collector","namespace":"fluent"}, "namespace": "fluent", "name": "fluent-bit-collector", "reconcileID": "b0128934-0ce5-4a21-9105-aa7df920cddd", "error": "error when handling finalizer: resource name may not be empty"}"

steps to reproduce:
1. create collector
2. delete it
3. check logs from operator

File: controllers/collector_controller.go [hunk 1]
```diff
@@ -170,17 +170,32 @@ func (r *CollectorReconciler) mutate(obj client.Object, co fluentbitv1alpha2.Col
 
 func (r *CollectorReconciler) delete(ctx context.Context, co *fluentbitv1alpha2.Collector) error {
 	var sa corev1.ServiceAccount
-	if err := r.Delete(ctx, &sa); err != nil && !errors.IsNotFound(err) {
+	err := r.Get(ctx, client.ObjectKey{Namespace: co.Namespace, Name: co.Name}, &sa)
```
[the remark this answers is not in the document]
Comment (ksdpmx) [reply controllers/collector_controller.go:173]: it seems like other controllers like fluentd have to update as well.
btw, does it make sense to check the owner reference before deletion? what if there's another resource which is not managed by this operator (i.e. fluentbit collector) with the same name?

File: controllers/collector_controller.go [hunk 2]
```diff
@@ -92,95 +93,122 @@ func (r *CollectorReconciler) Reconcile(ctx context.Context, req ctrl.Request) (
...
+	if err := r.Delete(ctx, &sts); err != nil && !errors.IsNotFound(err) {
 		return err
 	}
 
-	var sts appsv1.StatefulSet
-	if err := r.Delete(ctx, &sts); err != nil && !errors.IsNotFound(err) {
+	svc := corev1.Service{
+		ObjectMeta: metav1.ObjectMeta{
+			Name:      co.Name,
+			Namespace: co.Namespace,
```
Comment (wenchajun) [inline controllers/collector_controller.go:208]: https://github.com/fluent/fluent-operator/blob/master/pkg/operator/collector-service.go#L35
The name of the service does not have to be `co.name`

Comment (ksdpmx) [reply controllers/collector_controller.go:208]: correct. i'll fix it.

**Comment by benjaminhuo** [comment]
@wanjunlei are you ok with @ksdpmx‘s new change?

**Review by wanjunlei — APPROVED** [review]

**Review by benjaminhuo — APPROVED** [review]

**Comment by benjaminhuo** [comment]
@ksdpmx Thank you for this big refactoring

**Pull request merged**